lib: dai: Fix dai_get_device

dbaluta opened the pull request:
dai_get_device is called using DAI-types from IPC definition but calls dai_config_get() which returnes a Zephyr DAI type.

So, in order to fix this, we introduce `dai_type_to_zephyr` which converts a DAI-type from IPC definition to a Zephyr DAI type, so that later we can compare types in the same space.

The current code works because the two DAI types are matching, but it is no longer the case with latest introduces types.

Fixes: https://github.com/thesofproject/sof/issues/9803

LaurentiuM1234 commented on src/lib/dai.c:
```diff
@@ -162,7 +198,8 @@ const struct device *dai_get_device(uint32_t type, uint32_t index)
 		if (dai_config_get(zephyr_dev[i], &cfg, dir))
 			continue;
-		if (cfg.type == type && cfg.dai_index == index)
+		z_type = dai_type_to_zephyr(type);
```
no need to do this in the `for` loop. Just do it before.

also, maybe an error message if the conversion fails?

kv2019i reviewed (commented):
Thanks @dbaluta , looks good, few minor notes but I think the main one to move out of for loop and then good to go.

dbaluta commented:
v2
- fixed comments from @LaurentiuM1234  @kv2019i 
- changed name to sof_dai_type_to_zephyr
- moved call to sof_dai_type_to_zephyr outside of for loop
